src/averageAges.js

```diff
@@ -54,8 +67,23 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
+
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person => (
+    onlyWithSon
+      ? people.find(mother => mother.name === person.mother)
+      && person.sex === 'm'
+      : people.find(mother => mother.name === person.mother)
+  ));
```

Comment:
unify filter to avoid repeating the logic
and make sure you don't do something similar somewhere else
```suggestion
  const children = people.filter(person => (
    people.some(mother => mother.name === person.mother)
    && (onlyWithSon ? person.sex === 'm' : true)
  ));
```

Reply:
Somehow you ignore this. well, at least consider for the future.

Reply:
ohh pardon me I did not notice that I did not fix it ! thank you so much for your attention !